src/discord/discord.service.ts

```diff
@@ -94,12 +98,11 @@ export class DiscordService implements OnModuleInit {
   async processInteraction(interaction: Interaction) {
     if (!interaction.isChatInputCommand()) return;
 
-    const { commandName } = interaction;
-
-    const path = `${__dirname}/../slashCommand/commands/${commandName}.command.js`;
-    const command: { default: BaseCommand } = await import(path);
+    const command = await this.slashCommandService.getCommand(
+      interaction.commandName,
+    );
 
-    await command.default.handleCommand(interaction);
+    await command?.handleCommand(interaction);
```

Comment:
Any reason to make this two methods? `handleInteraction` might be sufficient here and then we can do the above internally in the slash command module?

Reply:
I followed the pattern you suggested for messages, where we have onMessageCreate that catches errors and processMessage that contains all the logic.
Now I'm thinking we could move everything else as well and
replace `this.client.on('interactionCreate', this.onInteractionCreate);`
with `this.client.on('interactionCreate', this.slashCommandService.onInteractionCreate);`

Reply:
Have a look: https://github.com/dnt-frivillig/dnt-discord-bot/pull/7/commits/a73c61760623feda24a39e8fe8bb988b05ce6f9d